File: src/models/order.rs
//! 订单相关模型
//!
//! 对应数据库中的 orders 和 order_addons 表

use serde::{Deserialize, Serialize};
use sqlx::types::chrono::{NaiveDateTime, NaiveDate};

/// 订单模型
/// 对应 orders 表
/// 
/// orders 表存储系统中的所有订单信息
/// 订单状态流转: pending -> confirmed -> assigned -> ongoing -> completed/cancelled
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Order {
    /// 订单编号 (主键)
    /// 格式: YYYYMMDDXXXXXX (日期+6位自增序列)
    pub order_id: String,
    
    /// 客户ID (外键关联 users 表)
    pub customer_id: i32,
    
    /// 服务人员ID (外键关联 users 表，可为空)
    pub worker_id: Option<i32>,
    
    /// 地址ID (外键关联 user_addresses 表)
    pub address_id: i32,
    
    /// 服务项目ID (外键关联 services 表)
    pub service_id: i32,
    
    /// 优惠券ID (外键关联 coupons 表，可为空)
    pub coupon_id: Option<i32>,
    
    /// 服务日期
    pub service_date: NaiveDate,
    
    /// 时间段，枚举值:
    /// - "morning": 上午 (09:00-12:00)
    /// - "afternoon": 下午 (12:00-18:00)
    /// - "evening": 晚上 (18:00-22:00)
    /// - "full_day": 全天 (09:00-22:00)
    pub time_slot: String,
    
    /// 服务时长/数量
    pub duration: f64,
    
    /// 单价
    pub unit_price: f64,
    
    /// 小计金额
    pub subtotal: f64,
    
    /// 优惠金额
    pub discount_amount: f64,
    
    /// 总金额
    pub total_amount: f64,
    
    /// 支付状态，枚举值:
    /// - "pending": 待支付
    /// - "paid": 已支付
    /// - "refunded": 已退款
    pub payment_status: String,
    
    /// 订单状态，枚举值:
    /// - "pending": 待确认
    /// - "confirmed": 已确认
    /// - "assigned": 已指派
    /// - "ongoing": 进行中
    /// - "completed": 已完成
    /// - "cancelled": 已取消
    pub order_status: String,
    
    /// 特殊要求 (可选)
    pub special_instructions: Option<String>,
    
    /// 取消原因 (可选)
    pub cancellation_reason: Option<String>,
    
    /// 计划开始时间
    pub scheduled_start_time: Option<NaiveDateTime>,
    
    /// 实际开始时间
    pub actual_start_time: Option<NaiveDateTime>,
    
    /// 实际结束时间
    pub actual_end_time: Option<NaiveDateTime>,
    
    /// 订单创建时间
    pub created_at: NaiveDateTime,
    
    /// 订单更新时间
    pub updated_at: NaiveDateTime,
}

/// 订单附加服务模型
/// 对应 order_addons 表
/// 
/// order_addons 表存储订单选择的附加服务信息
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OrderAddon {
    /// 订单附加服务ID (主键)
    pub order_addon_id: i32,
    
    /// 订单ID (外键关联 orders 表)
    pub order_id: String,
    
    /// 附加服务ID (外键关联 service_addons 表)
    pub addon_id: i32,
    
    /// 数量
    pub quantity: i32,
    
    /// 单价
    pub unit_price: f64,
    
    /// 创建时间
    pub created_at: NaiveDateTime,
}

File: forntend/mock/order.js
// mock/order.js
const Mock = require('mockjs')

// 模拟订单状态
const orderStatuses = ['pending', 'confirmed', 'assigned', 'ongoing', 'completed', 'cancelled']
const paymentStatuses = ['pending', 'paid', 'refunded', 'failed']

// 模拟订单数据
const orders = []
for (let i = 0; i < 50; i++) {
  const customerId = Mock.mock('@integer(1, 20)')
  const serviceId = Mock.mock('@integer(1, 30)')
  
  orders.push(Mock.mock({
    order_id: i + 1,
    customer_id: customerId,
    service_id: serviceId,
    service_date: '@date',
    time_slot: '@pick(["morning", "afternoon", "evening", "full_day"])',
    duration: '@integer(1, 4)',
    unit_price: '@float(30, 500, 2, 2)',
    subtotal: '@float(30, 2000, 2, 2)',
    discount_amount: '@float(0, 100, 2, 2)',
    total_amount: function() {
      return parseFloat((this.subtotal - this.discount_amount).toFixed(2))
    },
    'payment_status|1': paymentStatuses,
    'order_status|1': orderStatuses,
    special_instructions: '@cparagraph(1)',
    address_id: '@integer(1, 10)',
    worker_id: '@integer(1, 10)',
    created_at: '@datetime',
    updated_at: '@datetime'
  }))
}

// 获取订单列表
Mock.mock('/orders', 'get', (options) => {
  // 解析查询参数
  const url = new URL(options.url, 'http://localhost')
  const customerId = url.searchParams.get('customer_id')
  const status = url.searchParams.get('status')
  const page = parseInt(url.searchParams.get('page')) || 1
  const pageSize = parseInt(url.searchParams.get('page_size')) || 10
  
  let filteredOrders = orders
  if (customerId) {
    filteredOrders = orders.filter(o => o.customer_id == customerId)
  }
  if (status) {
    filteredOrders = filteredOrders.filter(o => o.order_status === status)
  }
  
  // 分页处理
  const total = filteredOrders.length
  const start = (page - 1) * pageSize
  const end = start + pageSize
  const paginatedOrders = filteredOrders.slice(start, end)
  
  return {
    code: 200,
    msg: '获取成功',
    data: {
      orders: paginatedOrders,
      total: total,
      page: page,
      page_size: pageSize
    }
  }
})

// 获取订单详情
Mock.mock(/^\/orders\/\d+/, 'get', (options) => {
  const url = options.url
  const id = parseInt(url.match(/\/orders\/(\d+)/)[1])
  
  const order = orders.find(o => o.order_id === id)
  
  if (!order) {
    return {
      code: 404,
      msg: '订单不存在'
    }
  }
  
  return {
    code: 200,
    msg: '获取成功',
    data: order
  }
})

// 创建订单
Mock.mock('/orders', 'post', (options) => {
  const orderData = JSON.parse(options.body)
  
  const newOrder = Mock.mock({
    order_id: orders.length + 1,
    customer_id: orderData.customer_id,
    service_id: orderData.service_id,
    service_date: orderData.service_date,
    time_slot: orderData.time_slot,
    duration: orderData.duration,
    unit_price: orderData.unit_price,
    subtotal: orderData.subtotal,
    discount_amount: orderData.discount_amount,
    total_amount: function() {
      return parseFloat((this.subtotal - this.discount_amount).toFixed(2))
    },
    payment_status: 'pending',
    order_status: 'pending',
    special_instructions: orderData.special_instructions,
    address_id: orderData.address_id,
    worker_id: null,
    created_at: '@datetime',
    updated_at: '@datetime'
  })
  
  orders.unshift(newOrder)
  
  return {
    code: 201,
    msg: '订单创建成功',
    data: newOrder
  }
})

// 更新订单状态
Mock.mock(/^\/orders\/\d+\/status/, 'put', (options) => {
  const url = options.url
  const id = parseInt(url.match(/\/orders\/(\d+)\/status/)[1])
  const { status } = JSON.parse(options.body)
  
  const order = orders.find(o => o.order_id === id)
  
  if (!order) {
    return {
      code: 404,
      msg: '订单不存在'
    }
  }
  
  order.order_status = status
  order.updated_at = Mock.mock('@datetime')
  
  return {
    code: 200,
    msg: '订单状态更新成功',
    data: order
  }
})